server/handle_map_zxy.go

```diff
@@ -113,10 +117,17 @@ func (req HandleMapZXY) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	tile := slippy.NewTile(req.z, req.x, req.y, TileBuffer, tegola.WebMercator)
 
-	//	filter down the layers we need for this zoom
+	// filter down the layers we need for this zoom
 	m = m.FilterLayersByZoom(req.z)
 
-	//	check for the debug query string
+	// We filtered out all the layers.
+	if len(m.Layers) == 0 {
+		log.Infof("map (%v) has no layers, at zoom %v", req.mapName, req.z)
+		http.Error(w, fmt.Sprintf("map (%v) has no layers, at zoom %v", req.mapName, req.z), http.StatusBadRequest)
```

Comment:
Yeah, I'm not sure. Here, seems like @pnorman thinks it should be 400, but it is more like a 404, but this would mean that if a map or layer name that is requested is not defined those should be 404s as well. In #375 we should discuss this.